ark/mibi/tiling_utils.py

```diff
@@ -446,22 +474,672 @@ def create_tiled_regions(tiling_params, moly_point, tma=False):
             fov = copy.deepcopy(tiling_params['fovs'][region_index])
             fov['centerPointMicrons']['x'] = cur_x
             fov['centerPointMicrons']['y'] = cur_y
-            fov['name'] = f'row{yi}_col{xi}'
+            fov['name'] = fov_names[index]
 
-            # append value to tiled_regions
-            tiled_regions['fovs'].append(fov)
+            # append value to fov_regions
+            fov_regions['fovs'].append(fov)
 
             # increment moly_counter as we've added another fov
             moly_counter += 1
 
             # append a moly point if moly_interval is set and we've reached the interval threshold
             if 'moly_interval' in tiling_params and \
                moly_counter % tiling_params['moly_interval'] == 0:
-                tiled_regions['fovs'].append(moly_point)
+                fov_regions['fovs'].append(moly_point)
 
         # append moly point to seperate runs if not last and if specified
         if tiling_params['moly_run'] == 'Y' and \
            region_index != len(tiling_params['region_params']) - 1:
-            tiled_regions['fovs'].append(moly_point)
+            fov_regions['fovs'].append(moly_point)
+
+    return fov_regions
+
+
+def convert_microns_to_pixels(coord):
+    """Convert the coordinate in stage microns to optical pixels
+
+    In other words, co-register using the centroid of a FOV.
+
+    The values are coerced to ints to allow indexing into the slide.
+    Coordinates are also returned in (y, x) form to account for a different coordinate axis.
+
+    Args:
+        coord (tuple):
+            The coordinate in microns to convert
+
+    Returns:
+        tuple:
+            The converted coordinate from microns to pixels
+    """
+
+    # NOTE: all conversions are done using the fiducials
+    # convert from microns to stage coordinates
+    stage_coord_x = (coord[0] * 0.001001 - 0.3116)
```

Comment:
Can these hard-coded constants be turned into variables with informative names? That way if the co-registration changes in the future it will be more obvious what needs to be updated here.